cogs/GuildSettings.py

```diff
@@ -501,141 +512,309 @@ def version_2_patch(all_guild_settings):
+                    secondary_rating_display_text TEXT,
+                    primary_rating_description_text TEXT,
+                    secondary_rating_description_text TEXT,
+                    primary_leaderboard_num_secondary_players INTEGER,
+                    secondary_leaderboard_num_secondary_players INTEGER,
+                    joining_time INTEGER,
+                    extension_time INTEGER,
+                    should_ping BOOLEAN,
+                    create_voice_channels BOOLEAN,
+                    send_scoreboard_text BOOLEAN,
+                    room_open_time INTEGER,
+                    lockdown_on BOOLEAN,
+                    created_channel_name TEXT,
+                    rating_command_on BOOLEAN,
+                    rating_command_primary_rating_embed_title TEXT,
+                    rating_command_secondary_rating_embed_title TEXT,
+                    show_rating BOOLEAN
+                    );""",
+        """CREATE TABLE IF NOT EXISTS roles_have_power (
+                    guild_id TEXT,
+                    role_name TEXT,
+                    FOREIGN KEY(guild_id) REFERENCES guild_settings(guild_id) ON DELETE CASCADE
+                    );""",
+        """CREATE TABLE IF NOT EXISTS roles_can_see_primary_leaderboard_rooms (
+                    guild_id TEXT,
+                    role_name TEXT,
+                    FOREIGN KEY(guild_id) REFERENCES guild_settings(guild_id) ON DELETE CASCADE
+                    );""",
+        """CREATE TABLE IF NOT EXISTS roles_can_see_secondary_leaderboard_rooms (
+                    guild_id TEXT,
+                    role_name TEXT,
+                    FOREIGN KEY(guild_id) REFERENCES guild_settings(guild_id) ON DELETE CASCADE
+                    );"""
+    ]
+
+    try:
+        for statement in sql_statements:
+            await con.execute(statement)
+        await con.commit()
+    except sql.Error as e:
+        print(f"Error creating tables: {e}")
+
     global GUILD_SETTINGS
     try:
-        with open("guildsettings_backup.pkl", "rb") as pickle_in:
-            try:
-                temp = p.load(pickle_in)
-                if temp == None:
-                    temp = defaultdict(lambda: GuildSettings())
-                GUILD_SETTINGS = temp
-            except:
-                print("Could not read in pickle for guildsettings_backup.pkl data.")
-                GUILD_SETTINGS = defaultdict(lambda: GuildSettings())
-    except:
-        print("guildsettings_backup.pkl does not exist, so no guild settings loaded in. Will create when a guild makes settings.")         
-        GUILD_SETTINGS = defaultdict(lambda: GuildSettings())
-        
+        res = await con.execute("SELECT * FROM guild_settings")
+        rows = await res.fetchall()
+        for row in rows:
+            guild_id = row[0]
+            guild_settings = GuildSettings()
+            guild_settings.set_guild_id(int(guild_id))
+            guild_settings.primary_rating_command = row[1]

```

Comment:
I forget if the sqlite3 library allows you to do this, but most interfaces allow you to access the item using its name: ```row.primary_rating_command``` or ```row["primary_rating_command"]```